File: Server/src/main/java/lk/wnl/wijeya/entity/AgentOrderProduct.java
package lk.wnl.wijeya.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.math.BigDecimal;

@Getter
@Setter
@Entity
@Table(name = "agent_order_product")
public class AgentOrderProduct {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Integer id;

    @Column(name = "quantity")
    private Integer quantity;

    @Column(name = "line_total", precision = 10, scale = 2)
    private BigDecimal lineTotal;
    @JsonIgnore
    @NotNull
    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "agent_order_id", nullable = false)
    private AgentOrder agentOrder;

    @NotNull
    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "product_id", nullable = false)
    private Product product;

    @NotNull
    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "product_design_id", nullable = false)
    private ProductDesign productDesign;

}

File: Client/src/app/entity/agentorderproduct.ts
import {Product} from "./Product";
import {Productdesign} from "./productdesign";

export class AgentOrderProduct {
  id?: number;
  quantity?: number;
  product?: Product;
  lineTotal?: number;
  productDesign?: Productdesign;

  constructor(id?: number, product?: Product, productDesign?:Productdesign,  quantity?: number,lineTotal?: number) {
    this.id = id;
    this.quantity = quantity;
    this.product = product;
    this.lineTotal = lineTotal;
    this.productDesign = productDesign;
  }


}
